src/main.ts

```diff
@@ -57,15 +60,42 @@ async function run(): Promise<void> {
 
     // Pick the best project ID.
     if (!projectID) {
-      if (credentialsJSON && isServiceAccountKey(credentialsJSON)) {
+      if (clusterName.projectID) {
```

Comment:
I set the order of precedence here to be: 

1. Project ID as an input
1. Project ID in the resource name
2. Project ID from credentials
3. Environment variable